contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,291 @@
+    // rysk liquidity pool that this strategy is interacting with
+    IRyskLiquidityPool public immutable ryskLqPool;
+    // pending withdrawal receipt
+    IRyskLiquidityPool.WithdrawalReceipt pendingWithdrawal;
+
+    /**
+     * Emmited when a withdrawal has been initiated.
+     *
+     *@param shares to be withdrawn
+     */
+    event RyskWithdrawalInitiated(uint256 shares);
+
+    // rysk liquidity pool is 0x
+    error RyskLiquidityPoolCannotBe0Address();
+    // no withdrawal initiated
+    error RyskNoWithdrawalInitiated();
+    // cannot complete withdrawal in the same epoch
+    error RyskCannotComipleteWithdrawalInSameEpoch();
+
+    /**
+     * @param _vault address of the vault that will use this strategy
+     * @param _owner address of the owner of this strategy
+     * @param _ryskLiquidityPool address of the rysk liquidity pool that this strategy is using
+     * @param _underlying address of the underlying token
+     */
+    constructor(
+        address _vault,
+        address _owner,
+        address _ryskLiquidityPool,
+        address _underlying
+    ) {
+        if (_owner == address(0)) revert StrategyOwnerCannotBe0Address();
+        if (_ryskLiquidityPool == address(0))
+            revert RyskLiquidityPoolCannotBe0Address();
+        if (_underlying == address(0))
+            revert StrategyUnderlyingCannotBe0Address();
+        if (!_vault.doesContractImplementInterface(type(IVault).interfaceId))
+            revert StrategyNotIVault();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _owner);
+        _setupRole(MANAGER_ROLE, _vault);
+
+        vault = _vault;
+        ryskLqPool = IRyskLiquidityPool(_ryskLiquidityPool);
+        underlying = IERC20(_underlying);
+
+        underlying.approve(_ryskLiquidityPool, type(uint256).max);
+    }
+
+    //
+    // Modifiers
+    //
+
+    modifier onlyManager() {
+        if (!hasRole(MANAGER_ROLE, msg.sender))
+            revert StrategyCallerNotManager();
+        _;
+    }
+
+    //
+    // Ownable
+    //
+
+    /**
+     * Transfers ownership of the Strategy to another account,
+     * revoking previous owner's ADMIN role and setting up ADMIN role for the new owner.
+     *
+     * @notice Can only be called by the current owner.
+     *
+     * @param _newOwner The new owner of the contract.
+     */
+    function transferOwnership(address _newOwner)
+        public
+        override(Ownable)
+        onlyOwner
+    {
+        if (_newOwner == address(0x0)) revert StrategyOwnerCannotBe0Address();
+        if (_newOwner == msg.sender)
+            revert StrategyCannotTransferOwnershipToSelf();
+
+        _transferOwnership(_newOwner);
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newOwner);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Rysk strategy is asynchronous meaning it doesn't support immediate withdrawals.
+     *
+     * @return false always
+     */
```

Comment:
Isn't this documented in IStrategy?

Reply:
It is, but I wanted to be more specific in this case. I agree it may not be necessary but it doesn't hurt :slightly_smiling_face:

Reply:
bit doing /// @inheritdoc IStrategy should be enough, otherwise we can always update the IStrategy docs to make it clear if you think it's not good enough

Reply:
i'll change the comment the ///@inheritdoc IStrategy